neqo-transport/src/frame.rs

```diff
@@ -262,10 +262,13 @@ impl Frame {
         data: &[u8],
         fin: bool,
         space: usize,
-    ) -> (Frame, usize) {
-        let mut remaining = space - 1 - Encoder::varint_len(stream_id);
+    ) -> Option<(Frame, usize)> {
+        let mut remaining = space.saturating_sub(1 + Encoder::varint_len(stream_id));
         if offset > 0 {
-            remaining -= Encoder::varint_len(offset);
+            remaining = remaining.saturating_sub(Encoder::varint_len(offset));
+        }
+        if remaining == 0 {
```

Comment:
It might be worth splitting this change out so that it can be tested.  I don't see tests here.

Reply:
ok split out into #359.

Reply:
OK, we need this change included for this PR's tests to pass. So instead of #359 (I don't want to gate this PR on that one) I'm just going to open an issue to add tests for `Frame::new_stream`.